### PR TITLE
Beta.3

### PR DESCRIPTION
**Pull request by mmurilo**
* ssl certificate creation automated
*  Documentation improvement

### CODE REVIEW

**Review by remipichon — APPROVED** [review]

File: k8s-cluster/README.md [hunk 1]
```diff
@@ -12,16 +12,21 @@
   - [2.3 MongoDB](#23-mongodb)
   - [2.4 RabbitMQ](#24-rabbitmq)
   - [2.5 Redis](#25-redis)
+  - [2.6 Files and Images Storage](#26-Files-and-Images-Storage)
```
Comment (EvHaus) [inline k8s-cluster/README.md:15]:
```suggestion
  - [2.6 File and Image Storage](#26-File-and-Image-Storage)
```

File: k8s-cluster/README.md [hunk 2]
```diff
@@ -62,6 +67,29 @@ ZenHub makes use of 1 externally managed Redis instance. This Redis instance tha
 
 There are two additional Redis instanceses that will run inside the cluster via our configuration. You do not need to worry about those.
 
+### 2.6 Files and Images Storage
```
Comment (EvHaus) [inline k8s-cluster/README.md:70]:
```suggestion
### 2.6 File and Image Storage
```

File: k8s-cluster/README.md [hunk 3]
```diff
@@ -12,16 +12,21 @@
...
   - [2.5 Redis](#25-redis)
+  - [2.6 Files and Images Storage](#26-Files-and-Images-Storage)
 - [3. Configuration](#3-configuration)
   - [3.1 Docker Registry](#31-docker-registry)
   - [3.2 Resource Scaling](#32-resource-scaling)
   - [3.3 Ingress](#33-ingress)
 - [4. Deployment](#4-deployment)
   - [4.1 Sanity Check](#41-sanity-check)
   - [4.2 Application Check](#42-application-check)
+  - [4.3 TLS/SSL](#43-TLS/SSL)
```
Comment (EvHaus) [inline k8s-cluster/README.md:23]:
```suggestion
  - [4.3 TLS/SSL](#43-tls-ssl)
```

File: k8s-cluster/README.md [hunk 4]
```diff
@@ -12,16 +12,21 @@
...
+  - [4.4 Buckets](#44-Buckets)
```
Comment (EvHaus) [inline k8s-cluster/README.md:24]:
```suggestion
  - [4.4 Buckets](#44-buckets)
```

File: k8s-cluster/README.md [hunk 5]
```diff
@@ -62,6 +67,29 @@ ZenHub makes use of 1 externally managed Redis instance. This Redis instance tha
 
 There are two additional Redis instanceses that will run inside the cluster via our configuration. You do not need to worry about those.
 
+### 2.6 Files and Images Storage
+
+Zenhub will required two object storage buckets to store files and images attached to issues using Zenhub's webbapp.
```
Comment (EvHaus) [inline k8s-cluster/README.md:72]:
```suggestion
ZenHub requires two object storage buckets. One to store files (PDFs, Word documents, etc...) and another to store images and videos which are attached to issues using ZenHub's webapp.
```

File: k8s-cluster/README.md [hunk 6]
```diff
@@ -62,6 +67,29 @@ ZenHub makes use of 1 externally managed Redis instance. This Redis instance tha
...
+
+Resources required:
+
+- 2 buckets
+  * `files` bucket
+  * `images` bucket
+- IAM user access_key_id
```
Comment (EvHaus) [inline k8s-cluster/README.md:79]:
```suggestion
- IAM user `access_key_id`
```

File: k8s-cluster/README.md [hunk 7]
```diff
@@ -62,6 +67,29 @@ ZenHub makes use of 1 externally managed Redis instance. This Redis instance tha
...
+- IAM user access_key_secret
```
Comment (EvHaus) [inline k8s-cluster/README.md:80]:
```suggestion
- IAM user `access_key_secret`
```

File: k8s-cluster/README.md [hunk 8]
```diff
@@ -62,6 +67,29 @@ ZenHub makes use of 1 externally managed Redis instance. This Redis instance tha
...
+- Bucket policy or permissions allowing bucket `list` and objects `get` from K8s nodes
```
Comment (EvHaus) [inline k8s-cluster/README.md:81]:
```suggestion
- Bucket policy or permissions allowing bucket `list` and objects `get` from Kubernetes nodes
```

File: k8s-cluster/README.md [hunk 9]
```diff
@@ -62,6 +67,29 @@ ZenHub makes use of 1 externally managed Redis instance. This Redis instance tha
...
+
+> ⚠️ **NOTE:** At the moment only AWS S3 API is supported.
```
Comment (EvHaus) [inline k8s-cluster/README.md:83]:
```suggestion
> ⚠️ **NOTE:** At the moment, only AWS S3 API is supported for buckets. S3-compatible APIs (such as IBM Cloud's Object Storage) should also work.
```

File: k8s-cluster/README.md [hunk 10]
```diff
@@ -62,6 +67,29 @@ ZenHub makes use of 1 externally managed Redis instance. This Redis instance tha
...
+
+To access and write these objects Zenhub also require CLI/API crdentials ( access_key_id/access_key_secret or simmiliar ) for a IAM user with at least read and write access.
```
Comment (EvHaus) [inline k8s-cluster/README.md:85]:
```suggestion
To access and write these objects ZenHub also requires CLI/API credentials (`access_key_id`/`access_key_secret` or similar) for a IAM user with at least read and write access.
```

File: k8s-cluster/README.md [hunk 11]
```diff
@@ -62,6 +67,29 @@ ZenHub makes use of 1 externally managed Redis instance. This Redis instance tha
...
+
+> ⚠️ **NOTE:** At the moment only IAM credentials are supported. There is some work in progress to support roles.
```
Comment (EvHaus) [inline k8s-cluster/README.md:87]:
```suggestion
> ⚠️ **NOTE:** At the moment, only authentication via IAM credentials is supported. Support for role-based authentication is planned for a future release.
```

File: k8s-cluster/README.md [hunk 12]
```diff
@@ -62,6 +67,29 @@ ZenHub makes use of 1 externally managed Redis instance. This Redis instance tha
...
+
+IAM credetials are used by ZHE to write ( `put` ) objects and to create temporary pre-signed links.
```
Comment (EvHaus) [inline k8s-cluster/README.md:89]:
```suggestion
IAM credentials are used by ZenHub to write (`put`) objects and to create temporary pre-signed links.
```

File: k8s-cluster/README.md [hunk 13]
```diff
@@ -62,6 +67,29 @@ ZenHub makes use of 1 externally managed Redis instance. This Redis instance tha
...
+
+To read ( `get` ) images and allow users to see uploaded images embebed in the issues, the cluster nodes or network need to have propper access.
```
Comment (EvHaus) [inline k8s-cluster/README.md:91]:
```suggestion
To read (`get`) images and allow users to see uploaded images embedded in the issue page, the cluster nodes or network need to have proper access.
```

File: k8s-cluster/README.md [hunk 14]
```diff
@@ -175,9 +203,9 @@ spec:
 
 ### 3.3 Ingress
 
-By default, we don't make any assumptions about the type of Ingress that is used with the cluster. It will be your responsibility to expose ZenHub through your preferred Ingress. The only requirement from the application side is that your Ingress targets the `nginx-gateway` service on port 80.
+By default, we don't make any assumptions about the type of Ingress that is used with the cluster. It will be your responsibility to expose ZenHub through your preferred Ingress. The only requirement from the application side is that your Ingress targets the `nginx-gateway` service on port 443.
 
-The provided manifests exposes ZenHub behind a single ClusterIP service, listening on port 80. You will need to setup and configure HTTPS through your Ingress (SSL configuration is not within the scope of "ZenHub for Kubernetes").
+The provided manifests exposes ZenHub behind a single ClusterIP service, listening on port 443. You will need to setup and configure HTTPS through your Ingress (Public facing SSL configuration is not within the scope of "ZenHub for Kubernetes").
```
Comment (EvHaus) [inline k8s-cluster/README.md:208]:
```suggestion
The provided manifests exposes ZenHub behind a single ClusterIP service, listening on port 443. You will need to setup and configure HTTPS through your Ingress (public facing SSL configuration is not within the scope of "ZenHub for Kubernetes").
```

File: k8s-cluster/README.md [hunk 15]
```diff
@@ -261,54 +289,12 @@ To verify that you deployment was successful, you should be able to visit the Ze
...
-    files:
-      # certificate name must be tls.crt and certificate key must be tls.key
-      - <some_path>/tls.crt
-      - <some_path>/tls.key
-    type: "kubernetes.io/tls"
-```
+#### 4.3.1 Application entrypoint
 
-- a bash script to create a self signed certificate is provided [here](options/tls/ssl/self-signed-creator.sh) 
+The `nginx-gateway` entrypoint uses HTTPS by default and utilizes a self signed ssl certificate.
```
Comment (EvHaus) [inline k8s-cluster/README.md:294]:
```suggestion
The `nginx-gateway` entrypoint uses HTTPS by default and utilizes a self-signed SSL certificate.
```

File: k8s-cluster/README.md [hunk 16]
```diff
@@ -261,54 +289,12 @@ To verify that you deployment was successful, you should be able to visit the Ze
...
+The certificate is automatic created.
```
Comment (EvHaus) [inline k8s-cluster/README.md:295]:
```suggestion
The certificate is automatically created when the pod starts.
```

File: k8s-cluster/README.md [hunk 17]
```diff
@@ -321,13 +307,39 @@ TLS connection with Postgres utilizes the secret generator at the end of the [ku
 
 - the files can be store any were and referenced inthe code with the full path.
 
-**⚠️ NOTE**: TLS connection for MongoDB is still in progress
+> ⚠️ **NOTE:** TLS connection for MongoDB is still in progress
```
Comment (EvHaus) [inline k8s-cluster/README.md:310]:
```suggestion
> ⚠️ **NOTE:** TLS connection for MongoDB is not supported at this time, but is being actively worked on
```

File: k8s-cluster/README.md [hunk 18]
```diff
@@ -321,13 +307,39 @@ TLS connection with Postgres utilizes the secret generator at the end of the [ku
...
+
+### 4.4 Buckets
 
-### 4.4 Enviroment without buckets
+To configure ZHE to store uploaded images and files as object in buckets all this varibales need to provided in kustomization.yaml
```
Comment (EvHaus) [inline k8s-cluster/README.md:314]:
```suggestion
To configure ZenHub to store uploaded images and files as objects in buckets, the following variables need to provided in `kustomization.yaml`
```

File: k8s-cluster/README.md [hunk 19]
```diff
@@ -321,13 +307,39 @@ TLS connection with Postgres utilizes the secret generator at the end of the [ku
...
+  - bucket_region=<bucket_region>
+  - bucket_domain=<bucket_public_domain>
+  - files_bucket_name=<bucket_name>
+  - images_bucket_name=<bucket_name>
+secret:
+  - bucket_secret_access_key=<some-key>
+```
 
-**⚠️ NOTE**: Using this option the enviroment is fucntional but files can **not** be uploaded. A error message will be shown.
+> ⚠️ **NOTE:** At the moment only AWS S3 API is supported.
```
Comment (EvHaus) [inline k8s-cluster/README.md:327]:
```suggestion
```

Remove. This is already documented above under "2.6 Files and Images Storage"

File: k8s-cluster/README.md [hunk 20]
```diff
@@ -321,13 +307,39 @@ TLS connection with Postgres utilizes the secret generator at the end of the [ku
...
+> ⚠️ **NOTE:** At the moment only IAM credentials are supported. There is some work in progress to support roles.
```
Comment (EvHaus) [inline k8s-cluster/README.md:328]:
```suggestion
```

Remove. This is already documented above under "2.6 Files and Images Storage"

File: k8s-cluster/README.md [hunk 21]
```diff
@@ -321,13 +307,39 @@ TLS connection with Postgres utilizes the secret generator at the end of the [ku
...
+
+IAM credetials are used by ZHE to write ( `put` ) objects and to create temporary pre-signed links.
```
Comment (EvHaus) [inline k8s-cluster/README.md:330]: There's a lot of duplication between this section ("4.4 Buckets") and "2.6 Files and Images Storage" above. We should combine them together. I think we can just move all the things in 4.4 to 2.6.

File: k8s-cluster/README.md [hunk 22]
```diff
@@ -321,13 +307,39 @@ TLS connection with Postgres utilizes the secret generator at the end of the [ku
...
+- Bucket policy/permissions allowing bucket `list` and objects `get` to K8s cluster VPC S3 Endpoint
+- Bucket policy/permissions allowing bucket `list` and objects `get` to K8s nodes role
+
+#### 4.4.1 Enviroment without buckets
+
+> ⚠️ **NOTE:** This option will only be avalible during the `BETA` phase
+
+> ⚠️ **NOTE:** Using this option the enviroment is fucntional but files can **not** be uploaded. A error message will be shown.
 
 To create a test enviroment whitout buckets
```
Comment (EvHaus) [inline k8s-cluster/README.md:344]:
```suggestion
To create a test environment without buckets
```

File: k8s-cluster/README.md [hunk 23]
```diff
@@ -481,3 +492,14 @@ kustomize build . | kubectl apply -f-
 
 ##### 3. Finalize
 * store the updated `kustomization.yaml`
+
+## 6. Roadmap
+
+Future features in progress:
```
Comment (EvHaus) [inline k8s-cluster/README.md:498]:
```suggestion
ZenHub Enterprise On-Premise 3.0.0 is actively in development and is currently available as a beta release. Prior to launching our Release Candidate, we are planning to add the following features:
```

File: k8s-cluster/README.md [hunk 24]
```diff
@@ -481,3 +492,14 @@ kustomize build . | kubectl apply -f-
...
+
+- MongoDB/DocuemntDB TLS
```
Comment (EvHaus) [inline k8s-cluster/README.md:500]:
```suggestion
- TLS support for MongoDB/DocumentDB
```

File: k8s-cluster/README.md [hunk 25]
```diff
@@ -481,3 +492,14 @@ kustomize build . | kubectl apply -f-
...
+- Browsers extensions downlaod
```
Comment (EvHaus) [inline k8s-cluster/README.md:501]:
```suggestion
- Support for downloading and serving ZenHub browser extensions
```

File: k8s-cluster/README.md [hunk 26]
```diff
@@ -481,3 +492,14 @@ kustomize build . | kubectl apply -f-
...
+- Admin interface UI
```
Comment (EvHaus) [inline k8s-cluster/README.md:502]:
```suggestion
- An administration interface UI for some internal application configuration
```

File: k8s-cluster/README.md [hunk 27]
```diff
@@ -481,3 +492,14 @@ kustomize build . | kubectl apply -f-
...
+- previous ZHE versions migration scripts
```
Comment (EvHaus) [inline k8s-cluster/README.md:503]:
```suggestion
- Scripts for facilitate migration between 2.x.x and 3.x.x versions of ZenHub Enterprise On-Premise
```

File: k8s-cluster/README.md [hunk 28]
```diff
@@ -481,3 +492,14 @@ kustomize build . | kubectl apply -f-
...
+- support for Github Enterprise 3
```
Comment (EvHaus) [inline k8s-cluster/README.md:504]:
```suggestion
- Support for GitHub Enterprise Server 3.0.0
```

File: k8s-cluster/README.md [hunk 29]
```diff
@@ -481,3 +492,14 @@ kustomize build . | kubectl apply -f-
...
+- support IAM roles to authorize ZHE to write bucket objects
```
Comment (EvHaus) [inline k8s-cluster/README.md:505]:
```suggestion
- Support for IAM role authorization when writing objects to S3 buckets
```

**Review by EvHaus — CHANGES_REQUESTED** [review]
Some documentation updates.

File: k8s-cluster/README.md (hunk 21 above)
Comment (mmurilo) [reply k8s-cluster/README.md:328]: keeping both for now.. 
2.6 is explanation about the requirements and usage
4.4 instruction to setup and use

File: k8s-cluster/README.md [hunk 30]
```diff
@@ -12,16 +12,21 @@
   - [2.3 MongoDB](#23-mongodb)
   - [2.4 RabbitMQ](#24-rabbitmq)
   - [2.5 Redis](#25-redis)
+  - [2.6 File and Image Storage](#26-File-and-Image-Storage)
```
Comment (dylanmartyn) [inline k8s-cluster/README.md:15]:
```suggestion
  - [2.6 File and Image Storage](#26-file-and-image-storage)
```

File: k8s-cluster/README.md [hunk 31]
```diff
@@ -481,3 +490,14 @@ kustomize build . | kubectl apply -f-
...
 * store the updated `kustomization.yaml`
+
+## 6. Roadmap
+
+ZenHub Enterprise On-Premise 3.0.0 is actively in development and is currently available as a beta release. Prior to launching our Release Candidate, we are planning to add the following features:
+
+- TLS support for MongoDB/DocumentDB
+- Support for downloading and serving ZenHub browser extensions
+- An administration interface UI for some internal application configuration
+- Scripts for facilitate migration between 2.x.x and 3.x.x versions of ZenHub Enterprise On-Premise
```
Comment (dylanmartyn) [inline k8s-cluster/README.md:501]:
```suggestion
- Scripts to facilitate migration between 2.x.x and 3.x.x versions of ZenHub Enterprise On-Premise
```

**Review by dylanmartyn — CHANGES_REQUESTED** [review]
Just a couple small changes

File: k8s-cluster/kustomization.yaml [hunk 32]
```diff
@@ -5,11 +5,11 @@ kind: Kustomization
 namespace: <a_dedicated_namespace>
 
 resources:
-  - github.com/ZenHubHQ/zenhub-enterprise.git//k8s-cluster/base
-  - github.com/ZenHubHQ/zenhub-enterprise.git//k8s-cluster/options/gateway-buckets
+  - github.com/ZenHubHQ/zenhub-enterprise/tree/3.0.0-beta.3/k8s-cluster/base
+  - github.com/ZenHubHQ/zenhub-enterprise.git/tree/3.0.0-beta.3/k8s-cluster/options/gateway-buckets
```
Comment (mmurilo) [inline k8s-cluster/kustomization.yaml:9]:
```suggestion
  - github.com/ZenHubHQ/zenhub-enterprise/tree/3.0.0-beta.3/k8s-cluster/options/gateway-buckets
```

File: k8s-cluster/kustomization.yaml [hunk 33]
```diff
@@ -5,11 +5,11 @@ kind: Kustomization
...
   # optional Job to check configuration is setup properly (test databases/caches and storage access)
-  - github.com/ZenHubHQ/zenhub-enterprise.git//k8s-cluster/options/sanitycheck
-  # - github.com/ZenHubHQ/zenhub-enterprise.git//k8s-cluster/options/gateway-local-files # to use in an environment without buckets
+  - github.com/ZenHubHQ/zenhub-enterprise.git/tree/3.0.0-beta.3/k8s-cluster/options/sanitycheck
+  # - github.com/ZenHubHQ/zenhub-enterprise.git/tree/3.0.0-beta.3/k8s-cluster/options/gateway-local-files # to use in an environment without buckets
```
Comment (mmurilo) [inline k8s-cluster/kustomization.yaml:12]:
```suggestion
  - github.com/ZenHubHQ/zenhub-enterprise/tree/3.0.0-beta.3/k8s-cluster/options/sanitycheck
  # - github.com/ZenHubHQ/zenhub-enterprise/tree/3.0.0-beta.3/k8s-cluster/options/gateway-local-files # to use in an environment without buckets
```

**Pull request merged**